libs/payments/cart/src/lib/cart.service.ts

```diff
@@ -324,14 +326,18 @@ export class CartService {
       handleEligibilityStatusMap[eligibility.subscriptionEligibilityResult];
 
     if (args.promoCode) {
-      try {
-        await this.promotionCodeManager.assertValidPromotionCodeNameForPrice(
-          args.promoCode,
-          price,
-          currency
-        );
-      } catch (e) {
-        throw new CartInvalidPromoCodeError(args.promoCode);
+      if (cartEligibilityStatus === CartEligibilityStatus.UPGRADE) {
+        delete args.promoCode;
```

Comment:
issue: The `previewUpcoming` on line 310 is using `args.promoCode` even though it's being deleted here. That would result in potentially incorrect values in the upcomingInvoice.